tools/ports/__init__.py

```diff
@@ -393,10 +393,20 @@ def add_deps(node):
     add_deps(port)
 
 
-def handle_use_port_arg(settings, name):
+def handle_use_port_arg(settings, arg):
+  args = arg.split('?', 1)
+  name, options = args[0], args[1] if len(args) == 2 else None
   if name not in ports_by_name:
     utils.exit_with_error(f'Invalid port name: {name} used with --use-port')
   ports_needed.add(name)
+  if options:
+    port = ports_by_name[name]
```

Comment:
If we standardize of query string format for the option maybe we should just call `parse_qs` here and pass in the results?

Reply:
I would be more than happy to do that. I just didn't know where you were standing. I can see 2 paths:

1. treat "options" as a total black box (which is what I implemented), which leaves ports developers to decide what is best for their ports. For example you could imagine having: `--use-port=sdl2_image?jpg` and that would still work (although it is not a valid query string)
2. enforce that "options" must be a valid query string in which case it would make more sense for the parsing to happen in `__init.py__.handle_use_port_arg`. That means more changes on the emscripten side but ensures a consistent syntax across ports.'

It sounds like you are option for 2. which is totally fine with me. Just let me know.